src/modules/flight_mode_manager/tasks/AutoFollowTarget/FlightTaskAutoFollowTarget.hpp

```diff
@@ -37,63 +37,94 @@
-// Lowpass applied to the estimated position of the target
-// before using it as control input
-static constexpr float POSITION_FILTER_ALPHA =	1.5f;
+// [m/s] Speed deadzone for which, under this velocity, the target orientation
+// tracking will freeze, since orientation can be noisy in low velocities
+static constexpr float TARGET_SPEED_DEADZONE_FOR_ORIENTATION_TRACKING = 1.0;
 
-// Filter on setpoints for smooth cinematic experience:
-// Lowpass applied to the follow-me angle setting, to ensure
-// smooth and circular transitions between settings
-static constexpr float FOLLOW_ANGLE_FILTER_ALPHA = 3.0f;
+// << Orbit Angle control related constants >>
 
-// Filter on setpoints for smooth cinematic experience:
-// Lowpass applied to the actual NED direction how the drone is facing the target
-// regarless of the setting. Used for dynamic tracking angles when the target makes a turn
-static constexpr float DIRECTION_FILTER_ALPHA =	3.0f;
+// [m/s^3] Maximum Jerk setting for generating the Follow Target Orbit trajectory
+static constexpr float ORBIT_TRAJECTORY_MAX_JERK = 4.0;
 
-// Filter on setpoints for smooth cinematic experience:
-// Lowpass applied for ramping up / down velocity feedforward term.
-// This is to avoid aggressive jerks when the target starts moving, because
-// velocity feed-forward is not applied at all while the target is stationary.
-static constexpr float VELOCITY_FF_FILTER_ALPHA = 1.0f;
+// [m/s^2] Maximum acceleration setting for generating the Follow Target Orbit trajectory
+static constexpr float ORBIT_TRAJECTORY_MAX_ACCELERATION = 2.0;
+
+// << RC Adjustment related constants >>
+
+// [m/s] Speed with which the follow distance will be adjusted by when commanded with deflection via RC command
+static constexpr float FOLLOW_DISTANCE_USER_ADJUST_SPEED = 2.0;
+// [m] Maximum follow distance that can be set by user's RC adjustment
+static constexpr float FOLLOW_DISTANCE_MAX = 100.f;
+
+// [m/s] Speed with which the follow height will be adjusted by when commanded with deflection via RC command
+static constexpr float FOLLOW_HEIGHT_USER_ADJUST_SPEED = 1.5;
+// [m] Maximum follow height that can be set by user's RC adjustment
+static constexpr float FOLLOW_HEIGHT_MAX = 100.f;
+
+// [rad/s] Angular rate with which the follow distance will be adjusted by when commanded with full deflection via RC command
+static constexpr float FOLLOW_ANGLE_USER_ADJUST_SPEED = 1.5;
+
+// [s] Time window constant that gets multiplied to user adjustment speed, to calculate the
+// 'acceptable' error in orbit angle / height / distance. If the difference between the setpoint
+// and actual state of the drone is smaller than this error, RC adjustments get applied.
+// Prevents setpoints diverging from the vehicle's actual position too much
+static constexpr float USER_ADJUSTMENT_ERROR_TIME_WINDOW = 0.5f;
+
+// Deadzone on both +/- direction for normalized stick input (-1, +1) where user adjustment will be ignored
```

Comment:
I wonder if the deadzone is necessary. It probably does not hurt to have some minimal deadzone. But why does the orbit flight task not require one too? Because orbit is using an exponential of the stick value, thus mitigating noise around 0?

Reply:
That's true. I could def use the Expo! I added this since the follow angle adjustment was really sensitive (especially when I just want to change the follow distance : pitch stick), it would swivel around the follow angle (roll stick) quite a lot.

But Expo could help fix this!

Reply:
I would do a separate PR on adding new verbose functions for expo values as well (currently it's index based, and I would like to add : getPitchExpo(), and so on). Or, should I include it to this PR?

Reply:
Sounds good

Reply:
Done!

Reply:
https://github.com/PX4/PX4-Autopilot/pull/19591